pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -258,9 +258,9 @@ def save_checkpoint(self, trainer, unused: Optional = None):
         to handle correct behaviour in distributed training, i.e., saving only on rank 0.
         """
         if unused is not None:
-            rank_zero_warn(
+            rank_zero_deprecation(
```

Comment:
this should be done using `warning_cache` to avoid duplicate deprecation messages. or maybe you can add warning_cache.deprecate that just calls `rank_zero_deprecation`.

Reply:
let's make this in a separate pr :]